shift_detector/checks/FrequentItemRulesCheck.py

```diff
@@ -40,21 +23,38 @@ class FrequentItemsetCheck(Check):
         in this value other than that it seems sufficiently reasonable.
     """
 
-    def __init__(self, min_support=0.01, min_confidence=0.15):
+    def __init__(self, min_support=0.01, min_confidence=0.15, rule_limit=5):
         self.min_support = min_support
         self.min_confidence = min_confidence
+        self.rule_limit = rule_limit
 
-    def run(self, store: Store) -> FrequentItemsetReport:
+    def run(self, store: Store) -> Report:
         """
         Calculate frequent rules, compress them and create a
         FrequentItemsetReport.
         :param store: the Store
         :return: FrequentItemsetReport
         """
-        df1 = store.df1
-        df2 = store.df2
+        precalculation_result = store[FrequentItemsetPrecalculation()]
+        compressed_rules = precalculation_result['compressed_rules']
+        examined_columns = precalculation_result['examined_columns']
+
+        shifted_clumns = list()
+        information = defaultdict(list)
+        count = 0
+        for rulecluster in compressed_rules:
+            columns = []
+            for (key, val) in rulecluster.attributes:
+                columns.append(key)
+            shifted_clumns.append(columns)
+
+            information[str(columns)].append(rulecluster.__str__())
+            count += 1
+            if count == self.rule_limit:
+                break
+
+        # remove duplicates from list of lists
+        shifted_clumns.sort()
+        shifted_clumns = list(shifted_clumns for shifted_clumns, _ in itertools.groupby(shifted_clumns))
 
-        item_rules = fpgrowth.calculate_frequent_rules(df1, df2, self.min_support,
-                                                       self.min_confidence)
-        compressed_rules = rule_compression.compress_rules(item_rules)
-        return FrequentItemsetReport(compressed_rules)
+        return Report(examined_columns, shifted_clumns, information)
```

Comment:
Why is no explanation returned? Information will be treated as explanation because it comes third